src/java/org/apache/cassandra/cql3/Operations.java

```diff
@@ -47,11 +49,34 @@ public final class Operations implements Iterable<Operation>
      */
     private final List<Operation> staticOperations = new ArrayList<>();
 
+    private final List<ReferenceOperation> regularSubstitutions = new ArrayList<>();
+    private final List<ReferenceOperation> staticSubstitutions = new ArrayList<>();
```

Comment:
`ModificationStatement#requiresRead()` calls `operations.requiresRead()` now, so actually I think it would be okay to add them...but I think we had a quick chat about this offline around one of the failing tests in the feedback branch, `testRefAutoRead()`.

Reply:
Yep we talked, we likely need this to know what columns to include in the read so that when we run we have the value rather than `null`